### PR TITLE
Attempt to fix posix.fstat

### PR DESCRIPTION
**Pull request by Kyle-Kyle**
The issue here is that there is no good way to resolve the guest path of a file given its `SimFile` object at this moment.
The hierarchy here is `SimFileSystem` -> `SimMount` -> `SimFile`.
Given a file path, it's easy to lookup its `SimFile` by following this hierarchy.
However, given a `SimFile`, looking up a file path is not easy with current implementation since mount information is not stored in `SimFile`. There are `ident` and `name` attributes in `SimFile`, but they are both fake identifiers and shouldn't be used for path resolution theoretically. 
A good approach is to add a backward reference to `SimMount` in `SimFile`. I couldn't make it work by adding the logic in `mount.insert`. And also, it raises a design question: is a `SimFile` allowed to be mounted in multiple `SimMount`?

This patch makes `fstat` work, but it is still not a proper fix because "given a SimFile, how to look up its `SimMount`" is still not resolved. Currently, I'm basically assuming the mount is at `/`

### CODE REVIEW

**Comment by Kyle-Kyle** [comment]
Also, don't merge it now, if you think this patch is fine, then the same thing should be implemented in `archr` too

**Comment by Kyle-Kyle** [comment]
And also, I think I need to write a testcase for this

**Review by rhelmot — APPROVED** [review]

**Comment by ltfish** [comment]
Waiting for linting errors to be fixed before merging this PR.

**Comment by Kyle-Kyle** [comment]
ping @ltfish

**Comment by Kyle-Kyle** [comment]
I believe the CI breaks because of other issues. may be not related to the patch

**Comment by ltfish** [comment]
Merged. Thanks!

**Pull request merged**
